corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js

```diff
@@ -225,6 +246,9 @@ hqDefine("data_dictionary/js/data_dictionary", [
                         if (element.deleted() && !element.id) {
                             return;
                         }
+                        if (!element.hasChanges) {
+                            return;
+                        }
```

Comment:
I think this will mess up how we save the index for case properties.
The index isn't stored on the case property but added during this iteration.

Reply:
This should be addressed in ([06e0775](https://github.com/dimagi/commcare-hq/pull/34781/commits/06e0775767e6a1ec9da6fac423a27e4397b5e26f)). 

Additionally, I have also fixed an issue where changing the group of a case property wasn't being saved ([37b0afb](https://github.com/dimagi/commcare-hq/pull/34781/commits/37b0afba141c4bfc9f71ba1822ee217176d208cf)).